src/uuid.ts

```diff
@@ -0,0 +1,51 @@
+const UUIDPattern = /^(?:[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}|00000000-0000-0000-0000-000000000000)$/i;
+
+export function formatUUID(uuid: string): string {
+  uuid = uuid.replace(/^.+:/, '');
+  const result = uuid.match(/^(.{8})(.{4})(.{4})(.{4})(.{12})$/);
+  if (result) uuid = result.slice(1, 6).join('-');
+  return uuid;
+}
+
+export function parseUUID(uuid: string): Uint8Array {
+  uuid = formatUUID(uuid);
+
+  if (typeof uuid !== 'string') {
+    throw new Error('$uuid wrong type');
+  }
+  if (!UUIDPattern.test(uuid)) {
```

Comment:
two comments:
- generally we tend to name file global constants IN_UPPER_CASE, this one might be called `UUID_RX` or something?
- it seems weird to validate using a regular expression _after_ you format the value in L11 above, is that intentional?

Reply:
Can fix uppercasing. 

Yes it is intentional to format -> validate. The validator accepts "proper" uuid's and will throw an error if it's invalid. The formatter is a buffer to convert SPEC-compliant uuids, to transform them before an error is thrown.

Reply:
it's still unclear to me from reading the source why the UUID needs to be formatted before its parsed, can you explain here? Or even better, explain in the source so that future readers can gain from what you learned.

Reply:
In the description of this ticket:
> Includes a `formatUUID` function not provided by the UUID package, in order to format the [UUID in accordance with python's `uuid.UUID` method](https://github.com/mongodb/specifications/blob/master/source/extended-json.rst#user-content-special-rules-for-parsing-uuid-fields:~:text=Parsers%20MUST%20interpret%20the%20%24uuid%20key,between%20hex%20character%20groups%20(e.g.%20c8edabc3f7384ca3b68dab92a91478a3).), where the `urn:uuid ` prefix is stripped and `-` are added in if they are not there prior to `base64` encoding.

Reply:
I removed the format function -> https://github.com/mongodb/js-bson/pull/403#discussion_r496909847